### PR TITLE
Feature/ims issue template

### PR DESCRIPTION
**Pull request by chriku**
Create IMSIssue for IMS specific information on each issue. This requires that templates which can be edited by the users are marked specifically

Depends on #6

### CODE REVIEW

File: src/main/kotlin/gropius/model/template/IMSTemplate.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    """
+)
+class IMSTemplate(
+    name: String, description: String, templateFieldSpecifications: MutableMap<String, String>, isDeprecated: Boolean
+) : Template<IMS, IMSTemplate>(name, description, templateFieldSpecifications, isDeprecated) {
+
+    @NodeRelationship(SubTemplate.PART_OF, Direction.INCOMING)
+    @GraphQLDescription("SubTemplate applied to all IMSProjects with this Template")
+    @delegate:Transient
+    val imsProjectTemplate by NodeProperty<IMSProject>()
```
Comment (nk-coding) [inline src/main/kotlin/gropius/model/template/IMSTemplate.kt:25]: I think this is the wrong type

File: src/main/kotlin/gropius/model/template/IMSTemplate.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @NodeRelationship(SubTemplate.PART_OF, Direction.INCOMING)
+    @GraphQLDescription("SubTemplate applied to all IMSIssues with this Template")
+    @delegate:Transient
+    val imsIssueTemplate by NodeProperty<IMSIssue>()
```
Comment (nk-coding) [inline src/main/kotlin/gropius/model/template/IMSTemplate.kt:30]: I think this is the wrong type

File: src/main/kotlin/gropius/model/template/MutableTemplatedNode.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package gropius.model.template
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLIgnore
+
+@GraphQLDescription("Interface for all types which support templates that can be written by the user.")
+interface MutableTemplatedNode : TemplatedNode {
+    @GraphQLIgnore
+    private fun graphQLKotlinDummyFunction() {
+        // Dummy function to avoid GraphQL kotlin optimization weirdness
```
Comment (nk-coding) [inline src/main/kotlin/gropius/model/template/MutableTemplatedNode.kt:10]: maybe change this to a KDoc comment

File: src/main/kotlin/gropius/model/template/MutableTemplatedNode.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package gropius.model.template
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import com.expediagroup.graphql.generator.annotations.GraphQLIgnore
+
+@GraphQLDescription("Interface for all types which support templates that can be written by the user.")
```
Comment (nk-coding) [inline src/main/kotlin/gropius/model/template/MutableTemplatedNode.kt:6]: technically, not the template, but the templated fields can be written by the user

**Review by nk-coding — CHANGES_REQUESTED** [review]

File: src/main/kotlin/gropius/model/template/IMSTemplate.kt (hunk 1 above)
Comment (chriku) [reply src/main/kotlin/gropius/model/template/IMSTemplate.kt]: 627fa894d0487f997628d7d8f967a37420cce136?

File: src/main/kotlin/gropius/model/template/IMSTemplate.kt (hunk 2 above)
Comment (chriku) [reply src/main/kotlin/gropius/model/template/IMSTemplate.kt]: 627fa894d0487f997628d7d8f967a37420cce136?

File: src/main/kotlin/gropius/model/template/MutableTemplatedNode.kt (hunk 3 above)
Comment (chriku) [reply src/main/kotlin/gropius/model/template/MutableTemplatedNode.kt]: 627fa894d0487f997628d7d8f967a37420cce136?

File: src/main/kotlin/gropius/model/template/MutableTemplatedNode.kt (hunk 4 above)
Comment (chriku) [reply src/main/kotlin/gropius/model/template/MutableTemplatedNode.kt]: 627fa894d0487f997628d7d8f967a37420cce136?

**Review by ModellbahnFreak — COMMENTED** [review]
Looks ok. Had to fight a bit with GitHub to only show changes from this PR and not include the template PR by @nk-coding. But other than that:

WUHU Everything becomes templatable :wink:

**Review by nk-coding — APPROVED** [review]

**Pull request merged**